pkg/rpc/server/server.go

```diff
@@ -270,17 +270,39 @@ func (s *Server) handleHTTPRequest(w http.ResponseWriter, httpRequest *http.Requ
 		return
 	}
 
-	err := req.DecodeData(httpRequest.Body)
+	err := in.DecodeData(httpRequest.Body)
 	if err != nil {
-		s.writeHTTPErrorResponse(req, w, response.NewParseError("Problem parsing JSON-RPC request body", err))
+		s.writeHTTPErrorResponse(request.NewRequest(), w, response.NewParseError("Problem parsing JSON-RPC request body", err))
 		return
 	}
 
-	resp := s.handleRequest(req, nil)
-	s.writeHTTPServerResponse(req, w, resp)
+	resp := s.handleIn(in, nil)
```

Comment:
Then iterate over batch and collect responses here.

Reply:
I made a separate method for this (it calls `(*Server).handleRequest` now) because it allows to avoid code duplication between `http` and `ws` handlers. Do we want to distinguish them?